### ISSUE TITLE
Add an option to make banphrases work only in online/offline chat

### ISSUE THREAD

**Issue by zneix** [pajbot/pajbot#1178]
Something like a dropdown list from ASCII Protection module would be cool in some banphrases.
![Ascii protection](https://cdn.zneix.eu/eMOAsJc.png)

I assume that the API response should be also updated; I am also aware that with such change, bots like supibot would have to respect whether a certain banphrase is working in which mode of chat.

**Comment by RAnders00** [comment]
This was already suggested in #546 but we closed that issue down due to the maintenance mode and nobody seemingly taking on implementing the feature. We can keep this issue open if somebody is interested in implementing it, otherwise I think the right thing would be to close this one too. Please comment too @pajlada

**Comment by pajlada** [comment]
Yeah what @RAnders00 said - if someone is interested in implementing then that's good :+1: let us know and we'll assign the issue to you. If no one is interested in implementing this, we'll close it after a while.

**Comment by zneix** [comment]
My bad, I didn't check closed issues. I will try to look around and see if I get what it takes to add this thing, but I am not that good at python so I don't think I will risk asking to assign this to me :sweat_smile:

**Comment by alazymeme** [comment]
> Yeah what @RAnders00 said - if someone is interested in implementing then that's good 👍 let us know and we'll assign the issue to you. If no one is interested in implementing this, we'll close it after a while.

I was looking into this and randers suggested refactoring the banphrase module, using `enabledWhileOnline` and `enabledWhileOffline` as boolean options instead of the current banphrase enabling system. Would you prefer the refactored method? Or the additional field method?

**Comment by pajlada** [comment]
> > Yeah what @RAnders00 said - if someone is interested in implementing then that's good +1 let us know and we'll assign the issue to you. If no one is interested in implementing this, we'll close it after a while.
> 
> I was looking into this and randers suggested refactoring the banphrase module, using `enabledWhileOnline` and `enabledWhileOffline` as boolean options instead of the current banphrase enabling system. Would you prefer the refactored method? Or the additional field method?

So instead of there being a single `enabled` boolean value for a banphrase in the database, there would be an enum `state` type returning `EnabledWhileOnlineAndOffline`, `EnabledWhileOffline`, `EnabledWhileOnline`, and `Disabled`?
`EnabledWhileOnlineAndOffline` would mirror the previous `enabled = true` state, and `Disabled` would mirror the previous `enabled = false` state.

**Comment by pajlada** [comment]
either way, enum or double bool is fine by me. just investigate what works best in the refactored code

**Comment by ALazyMeme** [comment]
I'll get around to working on this when I'm next on holidays

**Comment by Mr0ll3** [comment]
My suggestion https://github.com/pajbot/pajbot2/issues/528 is not exactly the same as this but offline/online would be nice too!

**Comment by ALazyMeme** [comment]
> My suggestion [pajbot/pajbot2#528](https://github.com/pajbot/pajbot2/issues/528) is not exactly the same as this but offline/online would be nice too!

The refactor would be similar to: toggling both online and offline would disable the banphrase

**Comment by RAnders00** [comment]
@Mr0ll3 I believe you can already do what you want by using `!edit banphrase <ID> --disabled/- 
-enabled`

**Comment by RAnders00** [comment]
Ah sorry, disregard my comment above, it seems like `!edit banphrase` doesn‘t exist. I was thinking of `!edit command`/`!edit playsound`

**Comment by Mr0ll3** [comment]
> > My suggestion [pajbot/pajbot2#528](https://github.com/pajbot/pajbot2/issues/528) is not exactly the same as this but offline/online would be nice too!
> 
> The refactor would be similar to: toggling both online and offline off would disable the banphrase

The difference is that this would not be possible to do from chat right?

**Comment by ALazyMeme** [comment]
> > > My suggestion [pajbot/pajbot2#528](https://github.com/pajbot/pajbot2/issues/528) is not exactly the same as this but offline/online would be nice too!
> > 
> > 
> > The refactor would be similar to: toggling both online and offline off would disable the banphrase
> 
> The difference is that this would not be possible to do from chat right?

Not necessarily. We could setup so something like a disable parameter just turns both options off.
